aws-lc-rs-testing/benches/kem_benchmark.rs

```diff
@@ -0,0 +1,143 @@
+        KemConfig {
+            algorithm: Algorithm::from(algorithm),
+            public_key: Vec::from(public_key).into(),
+            secret_key: Vec::from(secret_key).into(),
+            ciphertext: Vec::from(ciphertext).into(),
+        }
+    }
+}
+
+mod aws_lc_rs_benchmarks {
+
+    use aws_lc_rs::kem;
+
+    use crate::{Algorithm, KemConfig};
+    use kem::{PrivateKey, PublicKey, KYBER1024_R3, KYBER512_R3, KYBER768_R3};
+
+    fn algorithm(config: &KemConfig) -> &'static kem::Algorithm {
+        match config.algorithm {
+            Algorithm::KYBER512_R3 => &KYBER512_R3,
+            Algorithm::KYBER768_R3 => &KYBER768_R3,
+            Algorithm::KYBER1024_R3 => &KYBER1024_R3,
+        }
+    }
+
+    pub fn new_private_key(config: &KemConfig) -> PrivateKey {
+        PrivateKey::new(algorithm(config), &config.secret_key).unwrap()
+    }
+
+    pub fn new_public_key(config: &KemConfig) -> PublicKey {
+        PublicKey::new(algorithm(config), &config.public_key).unwrap()
+    }
+
+    pub fn keygen(config: &KemConfig) {
+        let private_key = PrivateKey::generate(algorithm(config)).unwrap();
+        let _public_key = private_key.compute_public_key().unwrap();
+    }
+
+    pub fn encapsulate(public_key: &PublicKey) {
+        public_key
+            .encapsulate((), |ct, ss| Ok((Vec::from(ct), Vec::from(ss))))
+            .unwrap();
+    }
+
+    pub fn decapsulate(config: &KemConfig, secret_key: &PrivateKey) {
+        let mut ciphertext = config.ciphertext.clone();
+        secret_key
+            .decapsulate(&mut ciphertext, (), |ss| Ok(Vec::from(ss)))
+            .unwrap();
+    }
+}
+
+fn bench_kem_keygen(c: &mut Criterion, config: &KemConfig) {
+    let bench_group_name = format!("KEM-{:?}-keygen", config.algorithm);
+
+    let mut group = c.benchmark_group(bench_group_name);
+
+    group.bench_function("AWS-LC", |b| {
+        b.iter(|| {
+            aws_lc_rs_benchmarks::keygen(config);
+        });
+    });
+}
+
+fn bench_kem_encapsulate(c: &mut Criterion, config: &KemConfig) {
+    let bench_group_name = format!("KEM-{:?}-encapsulate", config.algorithm);
+
+    let mut group = c.benchmark_group(bench_group_name);
+
+    group.bench_function("AWS-LC", |b| {
+        b.iter(|| {
+            let public_key = aws_lc_rs_benchmarks::new_public_key(config);
+            aws_lc_rs_benchmarks::encapsulate(&public_key);
+        });
+    });
+}
+
+fn bench_kem_decapsulate(c: &mut Criterion, config: &KemConfig) {
+    let bench_group_name = format!("KEM-{:?}-decapsulate", config.algorithm);
+
+    let mut group = c.benchmark_group(bench_group_name);
+
+    group.bench_function("AWS-LC", |b| {
+        b.iter(|| {
+            let private_key = aws_lc_rs_benchmarks::new_private_key(config);
+            aws_lc_rs_benchmarks::decapsulate(config, &private_key);
+        });
+    });
+}
+
+fn bench_kem(c: &mut Criterion) {
+    test::run(
+        test_file!("data/kem_benchmarks.txt"),
```

Comment:
why do you need KATs to perform benchmarks?

Reply:
Good question, not sure why this path was chosen in the prior PRs. I agree, will follow-up a change in a separate PR for review.